src/librustc/traits/error_reporting.rs

```diff
@@ -1112,7 +1112,8 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                 err.note("structs must have a statically known size to be initialized");
             }
             ObligationCauseCode::FieldSized => {
-                err.note("only the last field of a struct may have a dynamically sized type");
+                err.note("only the last field of a struct or an union may have a dynamically \
```

Comment:
The "last field" part is not relevant for unions, unions may not have unsized fields regardless of position.